File: Algorithms/14. Longest Common Prefix/golang/solution.vertical_scanning.go
func longestCommonPrefix(strs []string) string {
	if len(strs) == 0 {
		return ""
	}

	for i := 0; i < len(strs[0]); i++ {
		for _, str := range strs[1:] {
			if i == len(str) || str[i] != strs[0][i] {
				return strs[0][:i]
			}
		}
	}
	return strs[0]
}

File: Algorithms/14. Longest Common Prefix/python/solution.vertical_scanning.py
class Solution:
    def longestCommonPrefix(self, strs: list[str]) -> str:
        if len(strs) == 0 and strs is None:
            return ''
        for i in range(len(strs[0])):
            for string in strs[1:]:
                if i == len(string) or string[i] != strs[0][i]:
                    return strs[0][:i]
        return strs[0]
